raco/algebra.py

```diff
@@ -104,7 +104,7 @@ def collectParents(self, parent_map=None):
         if parent_map is None:
             parent_map = {}
         for c in self.children():
-            parent_map.setdefault(c, []).append(self)
+            parent_map.setdefault(id(c), []).append(self)
```

Comment:
I was following the comment and impl in https://github.com/uwescience/raco/blob/master/raco/backends/myria/myria.py#L1961 (also in function `call_compile_me()` later). I think... yes it looks like a bug to me.